### PR TITLE
added tutorial and unit tests for all exercises

### PR DESCRIPTION
**Pull request by duckmopsi**

### CODE REVIEW

File: src/ide/ide.ts [hunk 1]
```diff
@@ -2524,7 +2524,7 @@ export let ide = (function () {
 			e = createTypedEvent("canvas.KeyboardEvent", e);
 			module.interpreter.enqueueEvent("canvas.keyup", e);
 		});
-		/*
+
```
Comment (Jartreg) [inline src/ide/ide.ts:2527]: Was it intentional to uncomment this?

Comment (duckmopsi) [reply src/ide/ide.ts:2527]: Not sure tbh, I intended to only uncomment the code that's necessary for the tutorial.

Comment (Jartreg) [reply src/ide/ide.ts:2527]: Well, I'm asking because the title only says something about unit tests and this doesn't seem to be related...?

Comment (duckmopsi) [reply src/ide/ide.ts:2527]: Yes, I thought the tutorial had already been added but there hasn't been a PR to do that. Changed the title accordingly.

**Comment by TGlas** [comment]
Many many thanks!

**Pull request merged**